backend/src/dao/controller/member.controller.ts

```diff
@@ -0,0 +1,83 @@
+import { Body, Controller, Delete, Get, HttpCode, Param, Post, Put, UseGuards } from '@nestjs/common'
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger'
+import { Auth } from 'src/auth/auth.decorator'
+import { RolesGuard } from 'src/auth/auth.roles.guard'
+import { User } from 'src/auth/user.decorator'
+import { DaoAuth } from 'src/dao/dao.auth.decorator'
+import { Role } from 'src/user/user.dto'
+import { CreateMemberDto, ExtendedMemberDto, MemberDto, MemberInviteDto, MemberQueryDto } from '../../member/member.dto'
+import { MemberService } from '../../member/member.service'
+
+@ApiBearerAuth()
+@ApiTags('member')
+@Controller('/dao/:daoId/member')
+export class MemberController {
+  constructor(private memberService: MemberService) {}
+
+  @Post('list')
+  @HttpCode(200)
+  @DaoAuth(Role.founder, Role.projectManager)
+  list(@Param('daoId') daoId: string, @Body() query: MemberQueryDto): Promise<ExtendedMemberDto[]> {
+    query.daoId = daoId
+    return this.memberService.list(query)
+  }
+
+  @Post('invite')
+  @HttpCode(200)
+  @DaoAuth(Role.founder, Role.projectManager)
+  invite(
```

Comment:
@opny721 I need some clarification here. For the project management flow, we have the roles of:
- Project Manager: who creates a ProjectProposal (budget, deadline, KPIs) and submits to DAO for approval
- Contributor: who receives an invite from a Project Manager to take on a task.

Once a task is completed, Project Manager approves and DAO mints sweat tokens to Contributor in exchange for work deliverable.

I am getting confused with the code using terms as Members of the DAO. What do you consider Member of a DAO?

Reply:
A member of the DAO is technically a link between a user and a dao, roles such as Founder, PM are attached to a DAO membership and enforced duirng various operations. A member with no explicit role is a contributor to which a PM can assign tasks.